### PR TITLE
Add RE tests

### PR DESCRIPTION
**Pull request by ofekshenawa**

### CODE REVIEW

File: .github/workflows/test-redis-enterprise.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+name: RE Tests
+
+on:
+  push:
+    branches: [master, v9]
```
Comment (chayim) [inline .github/workflows/test-redis-enterprise.yml:5]: IMHO for now master only.. We shouldn't go back to v9 branches (yet). But yes, we will future them

File: .github/workflows/test-redis-enterprise.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  pull_request:
```
Comment (chayim) [inline .github/workflows/test-redis-enterprise.yml:6]: What about on pull_request (period). Maybe we should do the same platform label thing we have elsewhere, or is that overkill?

File: .github/workflows/test-redis-enterprise.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.21.x]
+        re-build: ["100.0.1-64402"]
```
Comment (chayim) [inline .github/workflows/test-redis-enterprise.yml:20]: awesome for future! :+1:

File: .github/workflows/test-redis-enterprise.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        # go-version: [1.19.x, 1.20.x, 1.21.x]
+
+    steps:
+      - name: Clone Redis EE docker repository
+        run: |
+          git clone https://github.com/RedisLabs/redis-ee-docker.git redis-ee
```
Comment (chayim) [inline .github/workflows/test-redis-enterprise.yml:26]: actions/checkout supports this

File: .github/workflows/test-redis-enterprise.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      - name: Install Docker-compose
+        run: |
+          DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
+          mkdir -p $DOCKER_CONFIG/cli-plugins
+          curl -SL https://github.com/docker/compose/releases/download/v2.23.3/docker-compose-linux-x86_64 -o $DOCKER_CONFIG/cli-plugins/docker-compose
```
Comment (chayim) [inline .github/workflows/test-redis-enterprise.yml:31]: docker-compose should already be installed, we rely on this being there in redis-py.

File: .github/workflows/test-redis-enterprise.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      - name: Build cluster
+        env:
+          IMAGE: "redislabs/redis-internal:${{ matrix.re-build }}"
```
Comment (chayim) [inline .github/workflows/test-redis-enterprise.yml:34]: Maybe instead we can load from dotnet *and* have a comment for why this exists? This way we set off fewer false alarms from automated security tools.

If not dotenv, at least a comment here IMHO.

File: .github/workflows/test-redis-enterprise.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+          RE_USERNAME: test@test.com
+          RE_PASS: 12345
+          RE_CLUSTER_NAME: test
+          OSS_CLUSTER: false
+          DB_PORT: 6379
+          DOCKER_ACCESS_TOKEN: ${{ secrets.DOCKER_ACCESS_TOKEN }}
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+        run: |
```
Comment (chayim) [inline .github/workflows/test-redis-enterprise.yml:42]:
```
name: <whatever>
run: ./build.sh
working-directory: redis-ee
```

Or similar

File: .github/workflows/test-redis-enterprise.yml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+          cd redis-ee/
+          ./build.sh
+
+      - name: Checkout repository
```
Comment (chayim) [inline .github/workflows/test-redis-enterprise.yml:46]: IMHO let's do both checkouts together at the top

File: .github/workflows/test-redis-enterprise.yml [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        uses: actions/checkout@v3
+
+      - name: Set up ${{ matrix.go-version }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Checkout code
```
Comment (chayim) [inline .github/workflows/test-redis-enterprise.yml:54]: Why another checkout?

**Comment by sjpotter** [comment]
dont understand why so many of the tests (all?) in commands_test.go are "non redis enterprise".  many of them seem like normal commands that should work fine.

**Comment by ofekshenawa** [comment]
@sjpotter, I have labeled tests as 'non-Redis Enterprise' when they involve commands unsupported in Redis Enterprise, such as ACL commands, config, etc., and tests that result in cross-slot violations

**Comment by sjpotter** [comment]
I'm surprised that commands like blpop/brpop aren't supported in redis enterprise?  but perhaps I should have?

**Comment by chayim** [comment]
IMHO we need this https://github.com/RedisLabs/redis-ee-docker/pull/7

**Comment by chayim** [comment]
Willing to pull in for ee-docker#7 as a separate item.

**Review by chayim — APPROVED** [review]

**Pull request merged**